core/src/history/events.rs

```diff
@@ -236,24 +241,20 @@ impl StableXOrderBookReading for EventRegistry {
         &self,
         block: BlockNumber,
     ) -> BoxFuture<Result<(AccountState, Vec<Order>)>> {
-        let (batch_id, block_number) = match block {
-            BlockNumber::Earliest => (BatchId(0), 0),
-            BlockNumber::Latest | BlockNumber::Pending => (BatchId::now(), u64::MAX),
+        let batch_id = match block {
+            BlockNumber::Earliest => BatchId(0),
+            BlockNumber::Latest | BlockNumber::Pending => BatchId::now(),
             // NOTE: Approximate the timestamp of the block by finding the batch
             // ID of the last event before the specified block.
-            BlockNumber::Number(block_number) => {
-                let block_number = block_number.as_u64();
-                let batch_id = self
-                    .events
-                    .range(bounds_until_end_of_block(block_number))
-                    .rev()
-                    .next()
-                    .map(|(_, Value { batch_id, .. })| *batch_id)
-                    .unwrap_or(BatchId(0));
-                (batch_id, block_number)
-            }
+            BlockNumber::Number(block_number) => self
+                .events
+                .range(bounds_until_end_of_block(block_number.as_u64()))
+                .rev()
+                .next()
+                .map(|(_, Value { batch_id, .. })| *batch_id)
+                .unwrap_or(BatchId(0)),
```

Comment:
We also want to replace this logic by looking up the batch number from the block-timestamp, right? But that will be a different PR I assume?

Reply:
That would require modifying the `EventRegistry` to have access to a `BlockTimestampReading` instance which is a pretty significant change. If we decide to do it, then it should be a separate PR.

That being said, I don't know how I feel about that change - currently the `EventRegistry` has a "pure" and nicely encapsulated logic for computing auction states just from past events, and adding a `BlockTimestampReading` would change that. I would argue that `UpdatingOrderbook` should be the "side-effect-y" implementation that updates the `EventRegistry` periodically and can query block timestamps and that `EventRegistry` doesn't necessarily need to implement `StableXOrderBookReading` as it has synchronous `auction_state_for_batch` and `acution_state_for_batch_at_block` methods that aren't async and build an auction state purely from past events.

Reply:
As discussed offline, maybe it's best if we unimplement this method for now, instead of having an incorrect implementation on which we might rely in the future...